src/Orleans.Runtime/Silo/Silo.cs

```diff
@@ -862,7 +854,7 @@ private void HandleProcessExit(object sender, EventArgs e)
             FastKill();
         }
 
-        private void UnobservedExceptionHandler(ISchedulingContext context, Exception exception)
+        private static void UnobservedExceptionHandler(ISchedulingContext context, Exception exception, ILogger logger)
```

Comment:
does this have to be passed into the handler? can't the handler just own this? and possible just rename the other to SiloUnobservedExceptionsHandler? I would move `DomainUnobservedExceptionHandler` related stuff there too

Reply:
I want to make this method static, so constructing a UnobservedExceptionsHandler class won't rely on constructing a Silo class first. 

As to "move DomainUnobservedExceptionHandler related stuff there too", this PR is about make UnobservedExceptionsHanlder class non static, so I don't think that is inside the scope of that. Let's make the scope stick to the PR's title. We can always refactor silo.cs in another PR.

Reply:
I wasn't implying we need to construct the silo first. I was suggesting not to pass any delegates, just have this utility log directly without any reference to Silo. I suggested renaming only because this is no longer used by the client, so the error message that is logged is only relevant when running inside a silo (not a reference to `Silo` class).
Regarding the Domain stuff... sure, it can be separate, but they belong in the same place, so that's why I'm saying that if the handler owns the log messages for unobserved task exceptions, it should also own the logging for the `AppDomain.UnhandledException` event. I find that refactoring very relevant for this PR, but sure, we can postpone fixing this inconsistency.